tests/conftest.py

```diff
@@ -77,7 +78,8 @@ def cleanup() -> None:
             def cleanup() -> None:
                 locale.setlocale(locale.LC_CTYPE, original_locale)
 
-            locale.setlocale(locale.LC_CTYPE, "en_US.UTF-8")
+            if original_locale is not None and original_locale[1] != "UTF-8":
+                locale.setlocale(locale.LC_CTYPE, "en_US.UTF-8")

```

Comment:
Interesting I would have assumed macOS would fail here as `original_locale` is `None` but obviously I must have miscalculated when this was actually needed as the tests are passing.

Reply:
Hmm, I think I meant to do `original_locale is None or ...` here. However, to be honest I'm surprised this even worked before. I've tried running the test suite with a different locale set, and the `setlocale()` call is insufficient to make Python work.

Perhaps the alternative would be to remove this entirely and rely on correct locale setting in the environment.

Reply:
We can certainly comment it all out and see what happens, if CI passes I'm happy. If it passes locally on your tests then we've removed an unneeded step in the tests which sounds like a plus.

Reply:
Ok, I'm going to try without it.